department-of-veterans-affairs/caseflow
# Limit VSO users' ability to view decision issues

Resolves [Jira ticket 1522](https://vajira.max.gov/browse/CASEFLOW-1522).

### Description

VSO users should be able to see request issues both before and after the request issues' respective decision dates.  However, VSO users should not be able to see decision issues until the issues' respective decision dates.  This change is meant to prevent VSO users from sharing information with veterans earlier than they should.

This PR also prevents VSO users from viewing decision issues' description field.  The description contains information that should not be accessed by users outside of board.

This PR did not change functionality for non-VSO users.

### Acceptance Criteria
- [x] VSO users can see request issues before and after their decision dates (there was no change to this functionality)
- [x] VSO users cannot see decision issues before their decision dates
- [x] VSO users can see decision issues after their decision dates
- [x] VSO users cannot see decision issues' description fields, regardless of the decision date


### Testing Plan
1. Make an appeal with a decision issue whose decision date has already passed.
```
FactoryBot.create(:appeal, :dispatched_with_decision_issue)
````
2. Log in as a VSO user and navigate to the above appeal's case details page.
3. Confirm that you can view the decision issue, but not the description field.
<img width="1768" alt="VSOUserViewPostDecisionDate" src="https://user-images.githubusercontent.com/80347702/123991475-c0edd300-d998-11eb-9e73-d4c2bc2c0c01.png">


4. Make an appeal with a decision issue whose decision date is in the future.
```
FactoryBot.create(:appeal, :decision_issue_with_future_date)
````

5. Confirm that you cannot view the decision issue while logged in as a VSO user.
<img width="1768" alt="VSOUserViewBeforeDecisionDate" src="https://user-images.githubusercontent.com/80347702/123992476-a5cf9300-d999-11eb-899a-a8cd1e985c6a.png">

6. Switch to a non-VSO user and navigate to the two previously viewed appeals.  Confirm that the change in functionality only affected the VSO users.
This appeal's decision issues have decision dates that have passed.
<img width="1768" alt="NonVSOUser" src="https://user-images.githubusercontent.com/80347702/123994361-50948100-d99b-11eb-86c8-891643c0b3f1.png">

This appeal's decision issues have decision dates that are in the future.
<img width="1768" alt="NonVSOUserDecisionDateInFuture" src="https://user-images.githubusercontent.com/80347702/123995069-f21bd280-d99b-11eb-95fd-a911e6804b04.png">


### User Facing Changes
 - [X] Screenshots of UI changes added to PR & Original Issue (See testing plan for screenshots)
